### ISSUE TITLE
Weird Linq Expression behaviour in NativeAOT

### ISSUE THREAD

**Issue by ryancheung** [dotnet/runtimelab#1262]
I  have a project depends on [LiteDB](https://github.com/mbdavid/LiteDB).

The minimal repro code:

```c#
namespace ConsoleApp1
{
    public sealed class KeyBindInfo
    {
        public int Id { get; set; }
        public string Category { get; set; }

        public KeyBindAction Action { get; set; }
    }

    public enum KeyBindAction
    {
        None,

        ConfigWindow,
    }

    class Program
    {
        public static ILiteCollection<KeyBindInfo> KeyBinds;

        static void Main(string[] args)
        {
            var mapper = new BsonMapper();

            mapper.Entity<KeyBindInfo>()
                .Id(x => x.Id, true);
            var UserDB = new LiteDatabase($"Filename=User.db;Mode=Exclusive", mapper);

            KeyBinds = UserDB.GetCollection<KeyBindInfo>();

            // This is working!
            var a = KeyBinds.Exists(x => x.Action == KeyBindAction.ConfigWindow);

            // This is NOT working!
            var tmp = KeyBindAction.ConfigWindow;
            a = KeyBinds.Exists(x => x.Action == tmp); // Crash in NativeAOT
            Console.WriteLine($"next............");

            Console.WriteLine("Done.");
        }
    }
}
```

The full repro project: https://github.com/ryancheung/NativeAOTRepro

**Comment by jkotas** [comment]
Stacktrace of the crash:
```
ConsoleApp1!S_P_CoreLib_System_ValueType::`vftable'
ConsoleApp1!S_P_CoreLib_System_Delegate__CreateObjectArrayDelegate+0x71 [/_/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Delegate.cs @ 396] 
ConsoleApp1!LiteDB_LiteDB_LinqExpressionVisitor__Evaluate+0x7e
ConsoleApp1!LiteDB_LiteDB_LinqExpressionVisitor__VisitBinary+0x157
ConsoleApp1!System_Linq_Expressions_System_Linq_Expressions_ExpressionVisitor__VisitLambda<System___Canon>+0x1a [/_/src/libraries/System.Linq.Expressions/src/System/Linq/Expressions/ExpressionVisitor.cs @ 346] 
ConsoleApp1!LiteDB_LiteDB_LinqExpressionVisitor__VisitLambda<System___Canon>+0x19
ConsoleApp1!LiteDB_LiteDB_LinqExpressionVisitor__Resolve+0x29
ConsoleApp1!LiteDB_LiteDB_LiteCollection_1<System___Canon>__Exists_2+0x29
```
The problem is that we do not see `Func<Int32>` delegate as constructed and skip emitting full vtable for it. The crash happens when we call the virtual GetThunk method on the delegate instance that is missing full vtable. Adding `GC.KeepAlive(new (() => 1));` to the Main method of the program makes the crash go away.

@MichalStrehovsky Could you please take a look?
`

**Comment by MichalStrehovsky** [comment]
Sounds like https://github.com/dotnet/corert/issues/7505. I didn't have a chance to look into it yet.

**Comment by MichalStrehovsky** [comment]
Reduced repro:

```csharp
using System;
using System.Runtime.CompilerServices;

var g = RuntimeHelpers.GetUninitializedObject(typeof(Gen<>).MakeGenericType(typeof(float)));

Console.WriteLine(g is Gen<float>);
Console.WriteLine(g.ToString());

class Gen<T>
{
    public override string ToString() => "Ola";
}
```

Basically, we can only hit this for delegates and `GetUninitializedObject` since the constructor would be missing for other scenarios.

**Comment by MichalStrehovsky** [comment]
Fixed in #1319.
